cli/azd/pkg/prompt/prompter.go

```diff
@@ -53,20 +53,20 @@ func NewDefaultPrompter(
 //
 // This currently means that subscription (AZURE_SUBSCRIPTION_ID) and location (AZURE_LOCATION) variables are set.
 func (p *DefaultPrompter) EnsureEnv(ctx context.Context) error {
-	if p.env.GetSubscriptionId() == "" {
+	if p.env.DotenvGet(environment.SubscriptionIdEnvVarName) == "" {
```

Comment:
Is this really the behavior we want? If the user has specified a shell/OS level env var / location shouldn't we use that vs prompting the user?

Maybe the real fix is to always set the env var value for location/subscription into the .env file regardless of its source.